jarbas/core/management/commands/reimbursements.py

```diff
@@ -30,21 +34,47 @@ def handle(self, *args, **options):
     @property
     def reimbursements(self):
         """Returns a Generator with a Reimbursement object for each row."""
-        with lzma.open(self.path, mode='rt') as file_handler:
-            for row in csv.DictReader(file_handler):
-                yield Reimbursement(**self.serialize(row))
+        print('Reading reimbursements dataset…')
+        reimbursements = pd.read_csv(self.reimbursements_path, dtype=object).fillna('')
+        
+        print('Reading irregularities dataset…')
+        irregularities = pd.read_csv(self.irregularities_path, dtype=object).fillna('')
+        
+        print('Merging datasets…')
+        on = ('applicant_id', 'year', 'document_id')
+        
+        for _, row in pd.merge(reimbursements, irregularities, on=on).iterrows():
+            yield Reimbursement(**self.serialize(row.to_dict()))
 
     def serialize(self, reimbursement):
         """Read the dict generated by DictReader and fix content types"""
-
-        missing = ('probability', 'suspicions')
-        for key in missing:
-            reimbursement[key] = None
+        
+        if 'probability' in reimbursement:
```

Comment:
I would split this logic here with a class method `to_float` (as we have `to_bool`, it looks like more coherent IMHO). This method could be fail proof returning `None` whenever there's no way to parse a value to `float`:

```python
    @staticmethod
    def to_float(value):
        try:
            return float(value)
        except TypeError:
            return None
```

Thus we can replace this `if`/`else` by

```python
    reimbursement['probability'] = self.to_float(reimbursement.get('probability'))
````